File: java/qiniu-lab-java/src/com/qiniulab/fop/VFrame.java
package com.qiniulab.fop;

import com.qiniu.common.QiniuException;
import com.qiniu.processing.OperationManager;
import com.qiniu.util.Auth;
import com.qiniu.util.StringMap;
import com.qiniu.util.UrlSafeBase64;
import com.qiniulab.config.LabConfig;

/**
 * 使用pfop接口来处理视频截图
 * 
 * vframe指令参数请参考
 * http://developer.qiniu.com/docs/v6/api/reference/fop/av/vframe.html
 * */

public class VFrame {

	public static void main(String[] args) {
		// 原文件信息
		String bucket = "if-pbl";
		String key = "mmm.mov";

		// 持久化结果保存信息
		String saveBucket = bucket;
		// 该结果名称不可以和原文件名称相同
		String saveKey = "mmm.jpg";
		String saveEntry = saveBucket + ":" + saveKey;

		Auth auth = Auth.create(LabConfig.ACCESS_KEY, LabConfig.SECRET_KEY);
		OperationManager m = new OperationManager(auth);

		// 持久化动作信息
		String fops = "vframe/jpg/offset/0|saveas/"
				+ UrlSafeBase64.encodeToString(saveEntry);
		// 私有队列名称，没有去后台创建
		String pipeline = "jemy";

		StringMap extraParams = new StringMap();
		// 持久化结果自动通知接收地址，可以不填
		extraParams.put("notifyURL", "http://www.abc.com/fop/notify");
		// 为了保障处理效率，请使用私有队列
		extraParams.put("pipeline", pipeline);
		try {
			String persistentId = m.pfop(bucket, key, fops, extraParams);
			// 然后根据persistentId查询结果
			System.out.println(persistentId);
		} catch (QiniuException e) {
			System.out.println(e.code());
			System.out.println(e.response.error);
		}
	}

}

File: python/qiniu-lab-python/video_fop/vframe.py
#coding=utf-8
__author__ = 'jemy'
import qiniu

'''
该例子演示如何从一个视频中截取某一时刻的帧。
详细文档：http://developer.qiniu.com/docs/v6/api/reference/fop/av/vframe.html
'''

accessKey = "<Your Access Key>"
secretKey = "<Your Secret Key>"


def vframe(format, offset, width, height, rotate, fromBucket, fromKey, usePipeline="", notifyUrl=None, saveBucket=None,
           saveKey=None):
    #prepare fop
    params = {
        "offset": offset,
        "w": width,
        "h": height,
        "rotate": rotate
    }
    fop = qiniu.build_op("vframe", format, **params)
    if saveBucket != None and saveKey != None:
        fop = qiniu.op_save(fop, saveBucket, saveKey)

    #do pfop
    auth = qiniu.Auth(accessKey, secretKey)
    pfop = qiniu.PersistentFop(auth, fromBucket, usePipeline, notifyUrl)
    retData, respInfo = pfop.execute(fromKey, [fop], force=None)
    if retData != None:
        print("PersistentId:" + retData["persistentId"])
    else:
        print("Error:")
        print("--StatusCode:" + str(respInfo.status_code))
        print("--Reqid:" + respInfo.req_id)
        print("--Message:" + respInfo.error)


def main():
    vframe("jpg", 0, 400, 300, "auto", "qiniu-lab", u"七牛云存储视频名片.mp4".encode("utf-8"), usePipeline="fff",
           saveBucket="qiniu-lab",
           saveKey="vframe_01.jpg")
    vframe("jpg", 10, 400, 300, 90, "qiniu-lab", u"七牛云存储视频名片.mp4".encode("utf-8"), usePipeline="fff",
           saveBucket="qiniu-lab",
           saveKey="vframe_02.jpg")
    vframe("png", 0, 400, 300, "270", "qiniu-lab", u"七牛云存储视频名片.mp4".encode("utf-8"), usePipeline="fff",
           saveBucket="qiniu-lab",
           saveKey="vframe_03.png")
    #save to different bucket
    vframe("png", 0, 400, 300, "180", "qiniu-lab", u"七牛云存储视频名片.mp4".encode("utf-8"), usePipeline="fff",
           saveBucket="if-pbl",
           saveKey="vframe_04.png")


if __name__ == "__main__":
    main()